backend/app.py

```
@@ -99,6 +102,22 @@ def create_app(config_class=None):
+    add_security_headers(app)
+    setup_error_handlers(app)
+
+    # Add security monitoring middleware
+    @app.before_request
+    def before_request():
+        try:
+            check_request_security()
+        except (SystemExit, KeyboardInterrupt):
+            # Re-raise critical exceptions
+            raise
+        except Exception as e:
```

Ruff (0.12.2)

Local variable `e` is assigned to but never used

Remove assignment to unused variable `e`

(F841)